File: 2022.01.28/python/1260.py
from collections import deque
n,m,v = map(int,input().split())
route = [[] for _ in range(n+1)]
for _ in range(m):
    a,b = map(int,input().split())
    route[a].append(b)
    route[b].append(a)
for i in range(1,n+1):
    route[i].sort()
visited = [ False for _ in range(n+1)]
result = ""
def dfs(a):
    global result
    visited[a] = True
    result += str(a)+' '
    for node in route[a]:
        if (visited[node]): continue
        dfs(node)
dfs(v)
print(result)
queue = deque()
queue.append(v)
visited = [ False for _ in range(n+1)]
visited[v] = True
result = ""
while queue:
    node = queue.popleft()
    result += str(node) +" "
    for node in route[node]:
        if (visited[node]): continue
        visited[node] = True
        queue.append(node)
print(result)
    

File: 2022.01.28/javascript/1260.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const [n,m,v] = input[0].split(' ').map(Number);
let a,b;
let route = [];
for ( let i = 0 ; i<=n ; i++){
  route.push([]);
}
for (let i = 1 ; i<=m ;  i++){
  [a,b] = input[i].split(' ').map(Number);
  route[a].push(b);
  route[b].push(a);
}
for (let i = 0 ; i<=n ; i++){
  route[i].sort((a,b) => a-b);
}

let visited = new Array(n+1).fill(false);
let result = [];
const dfs = (a) =>{
  result.push(a);
  visited[a] = true;
  for (let i = 0 ; i<route[a].length ; i++){
    if (visited[route[a][i]]) continue;
    dfs(route[a][i]);
  }
}
dfs(v);
console.log(result.join(" "));

q = [v];
result = [];
visited = new Array(n+1).fill(false);
visited[v] = true;
let nextNode;
const isEmptyArray = (arr)=> Array.isArray(arr) && arr.length === 0;
while (!isEmptyArray(q)){
  nextNode = q.shift();
  result.push(nextNode);
  for (let i = 0 ;i<route[nextNode].length ; i++){
    if (visited[route[nextNode][i]]) continue;
    visited[route[nextNode][i]] = true;
    q.push(route[nextNode][i]);
  }
}
console.log(result.join(" "));